smart_card_connector_app/src/window-apps-displaying.js

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This script implements displaying a list of connected
+ * applications in the Smart Card Connector App window.
+ */
+
+goog.provide('GoogleSmartCard.ConnectorApp.Window.AppsDisplaying');
+
+goog.require('GoogleSmartCard.DebugDump');
+goog.require('GoogleSmartCard.Logging');
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.PermissionsChecking.KnownApp');
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.PermissionsChecking.KnownAppsRegistry');
+goog.require('goog.array');
+goog.require('goog.asserts');
+goog.require('goog.dom');
+goog.require('goog.events.EventType');
+goog.require('goog.log.Logger');
+
+goog.scope(function() {
+
+/** @const */
+var GSC = GoogleSmartCard;
+
+/** @const */
+var PermissionsChecking = GSC.PcscLiteServerClientsManagement.PermissionsChecking;
+
+/** @const */
+var KnownApp = PermissionsChecking.KnownApp;
+
+/**
+ * @type {!goog.log.Logger}
+ * @const
+ */
+var logger = GSC.Logging.getScopedLogger('ConnectorApp.MainWindow');
+
+/**
+ * @type {Element}
+ * @const
+ */
+var appListElement = goog.dom.getElement('app-list');
+
+/**
+ * @const
+ */
+var knownAppsRegistry = new PermissionsChecking.KnownAppsRegistry();
+
+/**
+ * @type {goog.Promise.<!Array.<KnownApp>>}
+ */
+var lastKnownAppsPromise = null;
+
+/**
+ * @param {!goog.Promise.<!Array.<KnownApp>>} knownAppsPromise
+ * @param {!Array.<string>} appList
+ * @param {Array.<KnownApp>} knownApps
+ * @return {boolean}
+ */
+var updateAppView = function(knownAppsPromise, appList, knownApps) {
+    if (knownAppsPromise !== lastKnownAppsPromise) return false;
+
+    GSC.Logging.checkWithLogger(logger, !goog.isNull(appListElement));
+    goog.asserts.assert(appListElement);
+
+    goog.dom.removeChildren(appListElement);
+
+    for (var i = 0; i < appList.length; i++) {
+      var text = knownApps && knownApps[i] ?
+                 knownApps[i].name :
+                 '<' + appList[i] + '>';
+      var newElement = goog.dom.createDom('li', undefined, text);
+      goog.dom.append(appListElement, newElement);
+    }
+
+    return true;
+}
+
+/**
+ * @param {!Array.<string>} appList
+ */
+var onUpdateListener = function(appList) {
+  logger.fine('Application list updated, refreshing the view. ' +
+              'New list of id\'s: ' + appList);
+
+  var knownAppsPromise = knownAppsRegistry.tryGetByIds(appList);
+  lastKnownAppsPromise = knownAppsPromise;
+
+  knownAppsPromise.then(
+    function(knownApps) {
+      updateAppView(knownAppsPromise, appList, knownApps);
+    },
+    function(error) {
+      if (!updateAppView(knownAppsPromise, appList, null)) return;
+
+      logger.warning('Couldn\'t resolve appList: ' + error);
+    }
+  );
```

Comment:
nit: Indent all function arguments with 2 more spaces, and put the closing ")" bracket right after the last "}" bracket.
(see [https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting](https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting)).